introduction_to_applying_machine_learning/visual_object_detection/README.md

```diff
@@ -0,0 +1,145 @@
+
+!["sample2"](https://sagemaker-solutions-prod-us-east-2.s3.us-east-2.amazonaws.com/sagemaker-defect-detection/docs/sample2.png)
+
+The NEU surface defect database (see [references](#references)) is a *balanced* dataset which contains
+
+> Six kinds of typical surface defects of the hot-rolled steel strip are collected, i.e., rolled-in scale (RS), patches (Pa), crazing (Cr), pitted surface (PS), inclusion (In) and scratches (Sc). The database includes 1,800 grayscale images: 300 samples each of six different kinds of typical surface defects
+
+Here is a sample images of the six classes
+
+!["data sample"](https://sagemaker-solutions-prod-us-east-2.s3.us-east-2.amazonaws.com/sagemaker-defect-detection/docs/data.png)
+
+### 1.2. How to Prepare Your Data to Feed into the Model? <a name="preparedata"></a>
+
+There are data preparation and preprocessing steps and should be followed in the notebooks. It's critical to prepare your image annotations beforehand.
+For finetuning pretrained Sagemaker models, you need to prepare either a single `annotation.json` for all data, or a `RecordIO` file for both all images and all annotations. Check the notebook for details.
+
+### 1.3. What are the Outputs? <a name="output"></a>
+
+* For each image, the trained model will produce bounding boxes of detected visual defects (if any), the predicted defect type and prediction confidence score (0~1).
+* If you have a labeled test dataset, you could obtain the mean Average Precision (mAP) score for each model and compare among all the models.
+   * For example, the mAP scores on a test set of the NEU dataset
+
+     |     |  Type1 | Type1+HPO | Type2 | Type2+HPO|
+     | --- |  --- | --- | --- | ---|
+     | mAP | 0.067 | 0.226 | 0.371 | 0.375|
+
+
+### 1.4. What is the Estimated Cost? <a name="cost"></a>
+
+* Running the notebook costs around $130~140 USD. This notebook provides advanced materials, including finetuning two types of pretrained Sagemaker models **till convergence**, with and without hyperparameter optimization (HPO), and result in four models for inference. You could choose to train either one model, or all four models according to your budget and requirements. The cost and runtime for training each model (assuming using p3.2xlarge EC2 instance) are:
+
+   | Model | Cost (USD) | Runtime (Hours) | Billable time (Hours)|
+   |:----------:|:---------------:|:----:|:-----:|
+   |Type 1|    1.5     |       0.5       | 0.5|
+   |Type 1 with HPO (20 jobs)|    30.6    |       1*        | 10|
+   |Type 2|    4.6     |       1.5       | 1.5|
+   |Type 2 with HPO (20 jobs)|     92     |       3*        | 30|
+  (*) HPO tasks in this solution consider 20 jobs in total and 10 jobs in parallel. So 1 actual runtime hour amounts to 10 billable cost hours.
+* Please make sure you have read the cleaning up part [here](#cleaning-up) after training to avoid incurred cost from deployed models.
```

Comment:
Please use a more descriptive link (not "here").

Reply:
fixed, replaced with "read the cleaning up part in [Section 4](#cleaning-up) after ..."